Library/Rack.cpp

```diff
@@ -0,0 +1,69 @@
+#include "stdafx.h"
+#include "Rack.h"
+#include <iostream>
+#include <algorithm>
+
+Rack::Rack()
+{
+}
+
+void Rack::ViewAllBooks()
+{
+	if (list.size() > 0) {
+		unsigned int i = 1;
+		for (auto book : list)
+		{
+
+			std::cout << i << std::endl;
+			std::cout << book.book_id << std::endl;
+			std::cout << book.book_title << std::endl;
+			std::cout << book.author_name << std::endl;
+			std::cout << book.year << std::endl;
+			i++;
+		}
+	}
+	else {
+		std::cout << "There is no books in the library" << std::endl;
+	}
+}
+
+void Rack::AddBook(const Book &book)
+{
+	list.push_back(book);
```

Comment:
Here you push back reference to the book object. will it be valid when referenced variable becomes out of scope?